backend/api/views.py

```diff
@@ -120,6 +128,17 @@ class UserView(MultiSerializerViewMixin,
     def put(self, request, *args, **kwargs):
         return self.partial_update(request, *args, **kwargs)
 
+class UserListView(generics.ListAPIView):
+    queryset = User.objects.all()
+    serializer_class = UserSummarySerializer
+    filter_backends = (filters.SearchFilter,)
+    search_fields = ('username','first_name','last_name')
+    method_serializer_classes = {
```

Comment:
Since `UserListView` extends `generics.ListAPIView`, it only supports `HTTP GET`, so you can remove method_serializer_classes, it's not required. We're using it for views that requires different serializers for different methods.

Also, I don't know why you defined a search method here, but it isn't called from anywhere at all. I tested after removing this search method, and it works.